New resource - azurerm_integration_service_environment (#7763)
terraform-providers/terraform-provider-azurerm: website/docs/r/integration_service_environment.markdown

---
subcategory: "Logic App"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_integration_service_environment"
description: |-
  Manages a private and isolated Logic Apps instance within Azure virtual network.
---

# azurerm_integration_service_environment

> Manages a private and isolated Logic Apps instance within Azure virtual network.

Reviewer: ```suggestion
Manages private and isolated Logic App instances within an Azure virtual network.
```

Author: Thanks, will change it.

  address_prefixes     = ["10.0.1.0/26"]

  delegation {
    name = "integrationServiceEnvironments"
    service_delegation {
      name = "Microsoft.Logic/integrationServiceEnvironments"
    }
  }
}

resource "azurerm_subnet" "isesubnet2" {
  name                 = "isesubnet2"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.1.64/26"]
}

resource "azurerm_subnet" "isesubnet3" {
  name                 = "isesubnet3"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.1.128/26"]
}

resource "azurerm_subnet" "isesubnet4" {
  name                 = "isesubnet4"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.1.192/26"]
}

resource "azurerm_integration_service_environment" "example" {

  name                 = "example-ise"
  location             = azurerm_resource_group.example.location
  resource_group_name  = azurerm_resource_group.example.name

>   sku_name             = "Developer"
>   capacity             = 0

Reviewer: ```suggestion
  sku_name             = "Developer_0"
```

Author: Not sure if I follow here. Are you proposing to introduce new artificial SKU name? Would not be this kind of confusing? We have validation for SKU Developer and capacity 0 so we can catch bad input early - if this is the concern.

Author: Thanks, I changed it.

  access_endpoint_type = "Internal"
  virtual_network_subnet_ids = [
    azurerm_subnet.isesubnet1.id,
    azurerm_subnet.isesubnet2.id,
    azurerm_subnet.isesubnet3.id,
    azurerm_subnet.isesubnet4.id
  ]
  tags = {
    environment = "development"
  }
}
```

## Arguments Reference

The following arguments are supported:

> * `name` - (Required) The name which should be used for this Integration Service Environment. Changing this forces a new Integration Service Environment to be created.

Reviewer: ```suggestion
* `name` - (Required) The name of the Integration Service Environment. Changing this forces a new Integration Service Environment to be created.
```

Author: Thanks, will change it.

* `resource_group_name` - (Required) The name of the Resource Group where the Integration Service Environment should exist. Changing this forces a new Integration Service Environment to be created.

* `location` - (Required) The Azure Region where the Integration Service Environment should exist. Changing this forces a new Integration Service Environment to be created.

> * `sku_name` - (Required) The sku name of the Integration Service Environment. Possible Values are `Developer` and `Premium`. Changing this forces a new Integration Service Environment to be created.

Reviewer: ```suggestion
* `sku_name` - (Required) The sku name and capacity of the Integration Service Environment. Possible Values for `sku` element are `Developer` and `Premium` and possible values for the `capacity` element are from `0` to `10`.  Defaults to `sku` of `Developer` with a `Capacity` of `0` (e.g. `Developer_0`). Changing this forces a new Integration Service Environment to be created.
```

Author: Thanks, I changed it.

> * `capacity` - (Optional) Number of scale units for the Integration Service Environment. Min is `0`. Max is `10`. Defaults to `0`.

Reviewer: ```suggestion

```

Author: Thanks, I changed it.

> ~> **NOTE** For `sku_name` equal `Developer` capacity must be always `0`. For `sku_name` equal `Premium` capacity can be between `0` - `10`.

Reviewer: ```suggestion
~> **NOTE** For a `sku_name` using the `Developer` `sku` the `capacity` element must be always be `0`. For a `sku_name` using the `sku` of `Premium` the `capacity` element can be between `0` and `10`.
```

Author: Thanks, will change it
